Actions/AL-Go-Helper.ps1

```diff
@@ -201,38 +214,46 @@ function DownloadAndImportBcContainerHelper {
             }
         }
         if (Test-Path $repoSettingsPath) {
-            if (-not $BcContainerHelperVersion) {
-                $repoSettings = Get-Content $repoSettingsPath -Encoding UTF8 | ConvertFrom-Json | ConvertTo-HashTable
-                Write-Host $ENV:GITHUB_ACTION_PATH
-                $ap = "$ENV:GITHUB_ACTION_PATH".Split('\')
-                if ($ap -and $ap.Count -gt 4) {
-                    $folder = $ap[$ap.Count-5]
-                    $owner = $ap[$ap.Count-4]
-                    $repo = $ap[$ap.Count-3]
-                    $branch = $ap[$ap.Count-2]
-                    if ($folder -eq "_actions" -and $repo -eq "AL-Go-Actions") {
-                        if ($owner -eq "microsoft") {
-                            if ($branch -eq "preview") {
-                                $bcContainerHelperVersion = "preview"
-                            }
-                        }
-                        else {
-                            $bcContainerHelperVersion = "https://github.com/$owner/navcontainerhelper/archive/master.zip"
-                        }
+            $repoSettings = Get-Content $repoSettingsPath -Encoding UTF8 | ConvertFrom-Json | ConvertTo-HashTable
+            if (-not $bcContainerHelperVersion) {
+                if ($repoSettings.ContainsKey("BcContainerHelperVersion")) {
+                    $bcContainerHelperVersion = $repoSettings.BcContainerHelperVersion
+                    if ($bcContainerHelperVersion -like "https://*") {
+                        throw "Setting BcContainerHelperVersion to a URL is not allowed."
                     }
                 }
-                if ($bcContainerHelperVersion -eq "" -and $repoSettings.ContainsKey("BcContainerHelperVersion")) {
-                    $BcContainerHelperVersion = $repoSettings.BcContainerHelperVersion
-                    if ($BcContainerHelperVersion -like "https://*") {
-                        throw "Setting BcContainerHelperVersion to a URL is not allowed."
+            }
+            $ap = "$ENV:GITHUB_ACTION_PATH".Split([System.IO.Path]::DirectorySeparatorChar)
+            # Action Path under linux is something like: /home/runner/work/_actions/microsoft/AL-Go-Actions/main/WorkflowInitialize
+            # Action Path under Windows is something like: D:\a\_actions\freddydk\AL-Go-Actions\main\WorkflowInitialize
+            # ..../_actions/<owner>/AL-Go-Actions/branch/Action
+            # ..../   -5   /  -4   /     -3      /  -2  /  -1
+            # The code below checks if you are running a preview branch of AL-Go-Actions - then use Preview ContainerHelper
+            # If you are using a private fork of AL-Go-Actions - then use (if exists) a private fork of ContainerHelper as well
+            if ($ap -and $ap.Count -gt 4) {
+                $folder = $ap[$ap.Count-5]
+                $owner = $ap[$ap.Count-4]
```

Comment:
Wouldn't it better to use some of the GitHub environment variables, such as `GITHUB_ACTION_REPOSITORY` for the owner/repo name, `GITHUB_REF_NAME` for the branch name, `GITHUB_REPOSITORY_OWNER` for the repo owner?

The way GitHub Actions constructs these paths might change, right?

Reference: https://docs.github.com/en/actions/learn-github-actions/environment-variables

Reply:
I needed the branch from the Actions Repo - the github_ref_name doesn't give me that, neither does the github_repository_owner give me the owner of the Actions repo.

But... - the code goes away - I will use the GitHub_action_repository to determine the owner if containerhelperversion is set to private as the only special case.